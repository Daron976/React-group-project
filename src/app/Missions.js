import React, { useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { fetchMissons } from '../redux/missions/missions';
import MissionItems from './mission-items';

const Missions = () => {
  const dispatch = useDispatch();

  const missions = useSelector((state) => state.missionReducer);

  useEffect(() => {
    if (missions.length !== 10) {
      dispatch(fetchMissons());
    }
  });

  return (
    <section>
      <table>
        <tbody>
          <tr>
            <td><strong>Mission</strong></td>
            <td><strong>Description</strong></td>
            <td><strong>Status</strong></td>
            <td>&nbsp;</td>
          </tr>
          {missions.map((mission) => (
            <MissionItems
              key={mission.mission_id}
              name={mission.mission_name}
              description={mission.description}
            />
          ))}
        </tbody>
      </table>
    </section>
  );
};

export default Missions;
