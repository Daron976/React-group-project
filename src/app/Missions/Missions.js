import { useSelector } from 'react-redux';
import MissionItems from './mission-items';

const Missions = () => {
  const missions = useSelector((state) => state.missionReducer);

  return (
    <section>
      <table>
        <tbody>
          <tr>
            <td className="table-header"><strong>Mission</strong></td>
            <td className="table-header"><strong>Description</strong></td>
            <td className="table-header"><strong>Status</strong></td>
            <td className="table-header">&nbsp;</td>
          </tr>
          {missions.map((mission) => (
            <MissionItems
              key={mission.mission_id}
              name={mission.mission_name}
              description={mission.description}
              id={mission.mission_id}
              reserved={mission.reserved}
            />
          ))}
        </tbody>
      </table>
    </section>
  );
};

export default Missions;
