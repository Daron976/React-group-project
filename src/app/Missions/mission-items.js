import React, { useState } from 'react';
import PropTypes from 'prop-types';
import { useDispatch } from 'react-redux';
import { join, leave } from '../../redux/missions/missions';

const MissionItems = (props) => {
  const {
    id, name, description,
  } = props;

  const [btnState, setBtnState] = useState('Join Mission');

  const [memberState, setMemberState] = useState('NOT A MEMBER');

  const dispatch = useDispatch();

  const btnText = (e) => {
    if (e.target.textContent === 'Join Mission') {
      setBtnState('Leave Mission');
      setMemberState('Active Member');
      dispatch(join(id));
    } else {
      dispatch(leave(id));
      setMemberState('NOT A MEMBER');
      setBtnState('Join Mission');
    }
  };

  return (
    <tr>
      <td className="td-name">{name}</td>
      <td className="td-description">{description}</td>
      <td>
        <div
          className="td-status"
          style={{ background: memberState === 'NOT A MEMBER' ? '#6c767c' : '#24a1b3' }}
        >
          {memberState}
        </div>
      </td>
      <td className="td-btn">
        <button
          type="button"
          name="join"
          id="join"
          className="btn"
          onClick={btnText}
          style={{
            border: btnState === 'Join Mission' ? '1px solid #7a7a7a' : '1px solid #cf747c',
            color: btnState === 'Leave Mission' ? '#cf747c' : 'black',
          }}
        >
          {btnState}
        </button>
      </td>
    </tr>
  );
};

MissionItems.propTypes = {
  id: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired,
  description: PropTypes.string.isRequired,
};

export default MissionItems;
