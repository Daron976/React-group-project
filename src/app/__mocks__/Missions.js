const Missions = () => (
  <section>
    <table>
      <tbody>
        <tr>
          <td className="table-header"><strong>Mission</strong></td>
          <td className="table-header"><strong>Description</strong></td>
          <td className="table-header"><strong>Status</strong></td>
          <td className="table-header">&nbsp;</td>
        </tr>
      </tbody>
    </table>
  </section>
);
export default Missions;
