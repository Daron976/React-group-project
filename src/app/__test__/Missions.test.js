import renderer from 'react-test-renderer';
import Missions from '../__mocks__/Missions';

describe('Mission items', () => {
  test('Mission container items renders as expected', async () => {
    const snapMissionItems = renderer.create(
      <Missions />,
    )
      .toJSON();
    expect(snapMissionItems).toMatchSnapshot();
  });
});
