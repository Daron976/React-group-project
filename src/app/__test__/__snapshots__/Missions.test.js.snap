// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Mission items Mission container items renders as expected 1`] = `
<section>
  <table>
    <tbody>
      <tr>
        <td
          className="table-header"
        >
          <strong>
            Mission
          </strong>
        </td>
        <td
          className="table-header"
        >
          <strong>
            Description
          </strong>
        </td>
        <td
          className="table-header"
        >
          <strong>
            Status
          </strong>
        </td>
        <td
          className="table-header"
        >
          Â 
        </td>
      </tr>
    </tbody>
  </table>
</section>
`;
